src/ucp/core/ucp_mm.c

```diff
@@ -712,52 +714,53 @@ ucp_rndv_frag_malloc_mpools(ucs_mpool_t *mp, size_t *size_p, void **chunk_p)
         goto out;
     }
 
-    *chunk_p = chunk_hdr + 1;
-    ucs_mpool_chunk_get_info(mp->data, *size_p, (ucs_mpool_chunk_t*)*chunk_p,
-		             &elems, &num_elems);
-    ucs_debug("elems %p, num_elems %u", elems, num_elems);
+    num_elems =
+        ucs_mpool_num_elems_per_chunk(mp, (ucs_mpool_chunk_t*)(chunk_hdr + 1),
```

Comment:
no need to cast to ucs_mpool_chunk_t*

Reply:
chunk_hdr is of type ucp_rndv_frag_mp_chunk_hdr_t